From cd9cd8e4d4c210666075e957b022c2573caca931 Mon Sep 17 00:00:00 2001
From: jingpeiyang <jingpeiyang@eswincomputing.com>
Date: Thu, 16 Feb 2023 14:53:22 +0800
Subject: [PATCH] [riscv]Fix processor-bigint and processor failed.

By fix typo of https://chromium-review.googlesource.com/c/v8/v8/+/4178897 , two failed tests can pass:

1.test/mjsunit/tools/processor.mjs
2.test/mjsunit/tools/processor-bigint.mjs

Signed-off-by: jingpeiyang <jingpeiyang@eswincomputing.com>
Change-Id: I98435c389bbe6e3052b21c3307a571804b3a97c1
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4258672
Commit-Queue: Jakob Kummerow <jkummerow@chromium.org>
Reviewed-by: ji qiu <qiuji@iscas.ac.cn>
Reviewed-by: Jakob Kummerow <jkummerow@chromium.org>
Reviewed-by: Yahan Lu <yahan@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#85870}
---
 AUTHORS                                          | 1 +
 src/regexp/riscv/regexp-macro-assembler-riscv.cc | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/AUTHORS b/AUTHORS
index 21f0b915bb..515bb0d6cc 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -289,3 +289,4 @@ Zhongping Wang <kewpie.w.zp@gmail.com>
 柳荣一 <admin@web-tinker.com>
 Yang Xiang <xiangyangemail@gmail.com>
 Kotaro Ohsugi <dec4m4rk@gmail.com>
+Jing Peiyang <jingpeiyang@eswincomputing.com>
diff --git a/src/regexp/riscv/regexp-macro-assembler-riscv.cc b/src/regexp/riscv/regexp-macro-assembler-riscv.cc
index 7ffe6fe3ea..2e6df50c60 100644
--- a/src/regexp/riscv/regexp-macro-assembler-riscv.cc
+++ b/src/regexp/riscv/regexp-macro-assembler-riscv.cc
@@ -991,7 +991,7 @@ Handle<HeapObject> RegExpMacroAssemblerRISCV::GetCode(Handle<String> source) {
           .Build();
   Handle<InstructionStream> istream(code->instruction_stream(), isolate());
   LOG(masm_->isolate(),
-      RegExpCodeCreateEvent(Handle<AbstractCode>::cast(istream), source));
+      RegExpCodeCreateEvent(Handle<AbstractCode>::cast(code), source));
   return Handle<HeapObject>::cast(istream);
 }
 
-- 
2.35.1

